core/primitives/src/shard_layout.rs

```diff
@@ -49,7 +49,17 @@ use std::{fmt, str};
 
 pub type ShardVersion = u32;
 
-#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    serde::Serialize,
+    serde::Deserialize,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    ProtocolSchema,
```

Comment:
protocol schema is missing from the other structs - maybe good to add it there too